app/services/onlinetesting/ApplicationStatusCalculator.scala

```diff
@@ -41,11 +41,42 @@ trait ApplicationStatusCalculator {
     case _ => None
   }
 
+  //scalastyle:off cyclomatic.complexity
   private def sdipFaststreamCalc(phase: Phase, originalAppStatus: ApplicationStatus,
     evaluatedSchemes: List[SchemeEvaluationResult]): Option[ProgressStatus] = {
-    val results = evaluatedSchemes.filterNot(_.schemeId == SchemeId("Sdip")).map(s => Result(s.result))
-    faststreamCalc(phase, originalAppStatus, results)
+    val sdip = "Sdip"
+    val fsResults = evaluatedSchemes.filterNot(_.schemeId == SchemeId(sdip)).map(s => Result(s.result))
+    val fsOverallResult = faststreamCalc(phase, originalAppStatus, fsResults)
+
+    def phase1SdipFaststreamNotFailed(result: model.EvaluationResults.Result) =
+      result match {
+        case Amber => Some(PHASE1_TESTS_FAILED_SDIP_AMBER)
+        case _ => Some(PHASE1_TESTS_PASSED)
+      }
+
+    def phase2SdipFaststreamNotFailed(result: model.EvaluationResults.Result) =
+      result match {
+        case Amber => Some(PHASE2_TESTS_FAILED_SDIP_AMBER)
+        case _ => Some(PHASE2_TESTS_PASSED)
+      }
+
+    def phase3SdipFaststreamNotFailed(result: model.EvaluationResults.Result) =
+      result match {
+        case Amber => Some(PHASE3_TESTS_FAILED_SDIP_AMBER)
+        case _ => Some(PHASE3_TESTS_PASSED)
+      }
+
+    val sdipResult = evaluatedSchemes.filter(_.schemeId == SchemeId(sdip)).map(s => Result(s.result)).head
```

Comment:
From my point of view, arguably it will be more readable, in the sense that you do not have go back and forth to check the functions, the following code:


    val sdipResult = evaluatedSchemes.filter(_.schemeId == SchemeId(sdip)).map(s => Result(s.result)).head
     sdipResult match {
       case Amber if fsOverallResult.contains(PHASE1_TESTS_FAILED) => Some(PHASE1_TESTS_FAILED_SDIP_AMBER)
       case Green if fsOverallResult.contains(PHASE1_TESTS_FAILED) => Some(PHASE1_TESTS_PASSED)
       case Amber if fsOverallResult.contains(PHASE2_TESTS_FAILED) => Some(PHASE2_TESTS_FAILED_SDIP_AMBER)
       case Green if fsOverallResult.contains(PHASE2_TESTS_FAILED) => Some(PHASE2_TESTS_PASSED)
       case Amber if fsOverallResult.contains(PHASE3_TESTS_FAILED) => Some(PHASE3_TESTS_FAILED_SDIP_AMBER)
       case Green if fsOverallResult.contains(PHASE3_TESTS_FAILED) => Some(PHASE3_TESTS_PASSED)
       case Amber if fsOverallResult.contains(PHASE3_TESTS_PASSED) => Some(PHASE3_TESTS_PASSED_WITH_AMBER)
       case Green if fsOverallResult.isEmpty && phase == PHASE3 => Some(PHASE3_TESTS_PASSED_WITH_AMBER)
       case _ => fsOverallResult
     }

otherwise I think phase3SdipFaststreamNotFailed are a bit confusing because they are not verbs, I think is something about converting a result to an status, maybe toResultWhenPhase3SdipFaststreamNotFailed